node_modules/oae-content/lib/api.js

```diff
@@ -482,60 +483,73 @@ var _createContent = function(ctx, contentId, revisionId, resourceSubType, displ
         return callback(validator.getFirstError());
     }
 
-    // We need to check if we can interact with the additional members
-    // before we create the content item
-    var additionalMemberIds = _.keys(additionalMembers);
-    PrincipalsDAO.getPrincipals(additionalMemberIds, null, function(err, principals) {
+    // Get the principal ids, creating guest users if necessary    
+    PrincipalsAPI.getOrCreateGuestUsers(ctx, _.keys(additionalMembers), function(err, principalIds) {
```

Comment:
I can't really think of something good either. `getPrincipalsAndCreateGuestUsersWhenNecessary` :unamused: ?